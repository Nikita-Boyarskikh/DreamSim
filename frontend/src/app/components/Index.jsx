import React, { Suspense } from 'react';
import { hot } from 'react-hot-loader/root';

import { Provider as StoreProvider } from 'react-redux';
import { persistStore } from 'redux-persist';
import { PersistGate } from 'redux-persist/integration/react';

import DateMomentUtils from '@date-io/moment';
import { MuiPickersUtilsProvider } from '@material-ui/pickers';
import { MuiThemeProvider } from '@material-ui/core';
import CssBaseline from '@material-ui/core/CssBaseline';

import theme from 'app/styles/theme';
import moment from 'app/lib/moment';
import { isDebug } from 'app/lib/utils';
import { subscribe } from 'app/lib/i18n';
import store from 'app/state';

import Loader from 'app/components/Loader';
import Router from 'app/components/Router';

subscribe(store);
const persistor = persistStore(store);

let Index = () => (
  <StoreProvider store={store}>
    <PersistGate loading={<Loader />} persistor={persistor}>
      <MuiThemeProvider theme={theme}>
        <MuiPickersUtilsProvider utils={DateMomentUtils} libInstance={moment}>
          <CssBaseline />
          <Suspense fallback={<Loader />}>
            <Router />
          </Suspense>
        </MuiPickersUtilsProvider>
      </MuiThemeProvider>
    </PersistGate>
  </StoreProvider>
);

if (isDebug) {
  Index = hot(Index);
}

export default Index;
