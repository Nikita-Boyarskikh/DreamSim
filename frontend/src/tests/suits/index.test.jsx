import React from 'react';
import { render, cleanup } from '@testing-library/react';
import fetchMock from 'tests/mocks/fetch';

import Index from 'app/components/Index';

describe('Index', () => {
  afterEach(() => {
    cleanup();
    fetchMock.restore();
  });

  test('should renders without errors and warnings', async () => {
    const authKey = '1234567890-qwertyuiop-asdfghjkl-zxcvbnm';
    fetchMock.postOnce('/api/v1/auth/login/', {
      body: { key: authKey },
      headers: { 'content-type': 'application/json' }
    });

    const user = {
      id: 1,
      is_staff: true,
      username: 'durov',
      email: 'durov@vk.com',
      first_name: 'Павел',
      last_name: 'Дуров',
      patronymic: 'Валерьевич',
      group: 1,
      birthday: '1984-10-10',
      vk: 'https://vk.com/durov'
    };
    fetchMock.getOnce('/api/v1/user/current/', {
      body: user,
      headers: { 'content-type': 'application/json' }
    });

    const tool = { id: 1, name: 'TOOL', value: 'Tool name' };
    fetchMock.getOnce('/api/v1/enums/tools/', {
      body: [tool],
      headers: { 'content-type': 'application/json' }
    });

    const { asFragment } = render(<Index/>);
    expect(console.error).not.toHaveBeenCalled();
    expect(console.warn).not.toHaveBeenCalled();
    expect(asFragment()).toMatchSnapshot();
  });
});
