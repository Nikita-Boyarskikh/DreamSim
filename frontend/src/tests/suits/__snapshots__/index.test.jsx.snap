// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index should renders without errors and warnings 1`] = `
<DocumentFragment>
  <header
    class="MuiPaper-root MuiPaper-elevation4 MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary mui-fixed header"
    style="display: none;"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular header__toolbar"
    >
      <button
        class="MuiButtonBase-root MuiIconButton-root header__element header__menubutton"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <a
        aria-disabled="false"
        class="MuiButtonBase-root MuiIconButton-root header__element header__login"
        href="/login"
        role="button"
        tabindex="0"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M10.09 15.59L11.5 17l5-5-5-5-1.41 1.41L12.67 11H3v2h9.67l-2.58 2.59zM19 3H5c-1.11 0-2 .9-2 2v4h2V5h14v14H5v-4H3v4c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </a>
    </div>
  </header>
  <div
    class="PrivateSwipeArea-root-101 PrivateSwipeArea-anchorLeft-102"
    style="width: 20px; display: none;"
  />
  <div
    class="main page"
    style="display: none;"
  >
    <a
      aria-disabled="false"
      aria-label="Create scheme"
      class="MuiButtonBase-root MuiFab-root MuiFab-primary main__add-scheme"
      href="/editor"
      role="button"
      tabindex="0"
    >
      <span
        class="MuiFab-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
          />
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </a>
  </div>
  Loading...
</DocumentFragment>
`;
